python/ray/serve/backend_state_manager.py

```diff
@@ -151,7 +151,13 @@ def check_health(self) -> bool:
             self._health_check_ref = self._actor_handle.run_forever.remote()
 
         ready, _ = ray.wait([self._health_check_ref], timeout=0)
-        return len(ready) == 0
+
+        rst = len(ready) == 0
+        print(f"check_health -> {rst}")
+        print(f"ready: {ready}")
+        print(f"==== ray.get: {ray.get(ready)}")
```

Comment:
we should be doing this in `check_ready`, no? that way we can transition either to running or should_stop based on the result

Reply:
yep since ray.get is a blocking call until obj_ref is ready i think it only make sense to call it when we know for sure ray.wait returned us something. what you see in this commit has a bunch of debugging print statements while i was trying to figure out how health check works in ray serve ;)